ext/standard/streamsfuncs.c

```diff
@@ -67,10 +67,22 @@ PHP_FUNCTION(stream_socket_pair)
 		RETURN_FALSE;
 	}
 
-	array_init(return_value);
-
-	s1 = php_stream_sock_open_from_socket(pair[0], 0);
-	s2 = php_stream_sock_open_from_socket(pair[1], 0);
+    s1 = php_stream_sock_open_from_socket(pair[0], 0);
+    if (s1 == NULL) {
+        close(pair[0]);
+        close(pair[1]);
```

Comment:
Shouldn't the 2nd if close both pairs and the first if only pair[0]?

Reply:
I assume `php_stream_free(s1, ..` will close the underlying `pair[0]`

Reply:
Yes that is correct - `php_stream_free` will close the fd.